paths_cli/compiling/_gendocs/docs_generator.py

```diff
@@ -0,0 +1,146 @@
+import sys
+from paths_cli.compiling.core import Parameter
+from .json_type_handlers import json_type_to_string
+from .config_handler import DocCategoryInfo
+
+PARAMETER_RST = """* **{p.name}**{type_str} - {p.description}{required}\n"""
+
+
+class DocsGenerator:
+    """This generates the RST to describe options for compile input files.
+
+    Parameters
+    ----------
+    config : Dict[str, DocCategoryInfo]
+        mapping of category name to DocCategoryInfo for that category;
+        usually generated by :method:`.load_config`
+    """
+
+    parameter_template = PARAMETER_RST
+    _ANCHOR_SEP = "--"
+
+    def __init__(self, config):
+        self.config = config
+
+    def format_parameter(self, parameter, type_str=None):
+        """Format a single :class:`.paths_cli.compiling.Parameter` in RST
+        """
+        required = " (required)" if parameter.required else ""
+        return self.parameter_template.format(
+            p=parameter, type_str=type_str, required=required
+        )
+
+    def _get_cat_info(self, category_plugin):
+        cat_info = self.config.get(category_plugin.label, None)
+        if cat_info is None:
+            cat_info = DocCategoryInfo(category_plugin.label)
+        return cat_info
+
+    def generate_category_rst(self, category_plugin):
+        """Generate the RST for a given category plugin.
+
+        Parameters
+        ----------
+        category_plugin : :class:`.CategoryPlugin`
+            the plugin for which we should generate the RST page
+
+        Returns
+        -------
+        str :
+            RST string for this category
+        """
+        # TODO: move type_required to DocCategoryInfo (default True)
```

Comment:
this seems to be implemented?
```suggestion
```

Reply:
yep; in the last commit in this PR. Forgot to remove this TODO